.github/workflows/ci.yml

```diff
@@ -346,8 +346,8 @@ jobs:
         # Due to the relatively small number of usage_stats tests, we deem it appropriate to test them in their entirely through pytest
         run: pytest --no-sqlalchemy --aws-integration -v tests/integration/usage_statistics
   build-n-publish:
-    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
-    #if: github.event.push != null || github.event.schedule != null
+    needs: [doc-checks, static-analysis, docs-snippets, unit-tests, cloud-tests, marker-tests, py38-min-versions, py39-min-versions, py310-min-versions, import_gx, usage_stats_integration]
```

Comment:
require all of the other stages to pass before we `build-n-publish`

Reply:
nit: Can you format this like I did for `notify_on_failure`, ie one dependency per line.